src/section_list_cache.rs

```diff
@@ -0,0 +1,159 @@
+// Copyright 2016 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under (1) the MaidSafe.net Commercial License,
+// version 1.0 or later, or (2) The General Public License (GPL), version 3, depending on which
+// licence you accepted on initial access to the Software (the "Licences").
+//
+// By contributing code to the SAFE Network Software, or to this project generally, you agree to be
+// bound by the terms of the MaidSafe Contributor Agreement, version 1.1.  This, along with the
+// Licenses can be found in the root directory of this project at LICENSE, COPYING and CONTRIBUTOR.
+//
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.
+//
+// Please review the Licences for the specific language governing permissions and limitations
+// relating to use of the SAFE Network Software.
+
+use id::PublicId;
+use itertools::Itertools;
+use messages::GroupList;
+use routing_table::Prefix;
+use rust_sodium::crypto::sign::Signature;
+use std::collections::HashMap;
+use super::QUORUM;
+use super::XorName;
+
+pub type Signatures = HashMap<PublicId, Signature>;
+pub type PrefixMap<T> = HashMap<Prefix<XorName>, T>;
+
+pub struct SectionListCache {
+    // all signatures for a group list for a given prefix
+    signatures: PrefixMap<HashMap<GroupList, Signatures>>,
```

Comment:
I wonder whether it would be simpler (e.g. for pruning) to just have a `HashMap<(Prefix, GroupList), Signatures>`?